### PR TITLE
Display campaigns in feature card

### LINKED ISSUE #6665: Display campaigns to signup for on the onboarding experience
#### User Story

As a user on the first screen of onboarding I can sign up for other campaigns
#### Additional Information (optional)

Use solr to grab the campaigns
- display open, published campaigns
- boost staff pick campaigns
- boost if has a partner that is a sponsor  
- boost by `field_active_hours` (smallest values first) 

Do not display campaigns that user has already signed up for

### PR DESCRIPTION
**Pull request by itsjoekent**
#### What's this PR do?
- Displays campaigns in the feature card
- Lets users sign up for campaigns (displays quick animation for now, more advanced one to come next)
- Fixing a modal bug
- Fixing the on boarding experience being 100% width of the .chrome (it was really bothering me lol)
#### How should this be reviewed?

Does the feature card show all of the staff picked campaigns ranked by sponsor?
Does it show campaigns you've already signed up for? 
Are you able to signup for a campaign?
Can you click "Show me another"?
#### Any background context you want to provide?

👀 
#### Relevant tickets

Fixes #6665
and probably another one i forgot the # of
#### Checklist
- [ ] Tested on staging.

### CODE REVIEW

File: lib/themes/dosomething/paraneue_dosomething/js/components/FeatureCard.js [hunk 1]
```diff
@@ -1,4 +1,31 @@
 const React = require('react');
+const $ = require('jquery');
+
+const API_HOST = window.location.hostname;
+const API_PORT = window.location.port || undefined;
+const API_VERSION = '/api/v1/';
+const API_URL = `http://${API_HOST}${API_PORT ? ':' + API_PORT : ''}${API_VERSION}`;
+
+function checkForSponsor(campaign) {
+  var partners = campaign.affiliates.partners;
```
Comment (weerd) [inline lib/themes/dosomething/paraneue_dosomething/js/components/FeatureCard.js]: use `const`?

File: lib/themes/dosomething/paraneue_dosomething/js/components/FeatureCard.js [hunk 2]
```diff
@@ -1,4 +1,31 @@
...
+      return true;
+    }
+  }
+
+  return false;
+}
+
+function sortCampaigns(a, b) {
+  var aScore = a.time_commitment - (checkForSponsor(a) ? 1 : 0);
+  var bScore = b.time_commitment - (checkForSponsor(b) ? 1 : 0);
```
Comment (weerd) [inline lib/themes/dosomething/paraneue_dosomething/js/components/FeatureCard.js]: use `const`?

File: lib/themes/dosomething/paraneue_dosomething/js/components/FeatureCard.js [hunk 3]
```diff
@@ -7,9 +34,80 @@ const React = require('react');
...
+      current_campaign: '',
+      signup_animation: ''
+    };
+  }
+
+  componentDidMount() {
+    this.serverRequest = $.get(`${API_URL}campaigns?random=true&staff_pick=true`, result => {
+      // Filter out campaigns users have signed up for & then sort remaining campaigns
+      var userSignups = Drupal.settings.dsUser.activity.signups;
+      var campaigns = result.data.filter(campaign => userSignups.indexOf(parseInt(campaign.id)) == -1).sort(sortCampaigns);
```
Comment (weerd) [inline lib/themes/dosomething/paraneue_dosomething/js/components/FeatureCard.js]: `const`?

File: lib/themes/dosomething/paraneue_dosomething/js/components/FeatureCard.js [hunk 4]
```diff
@@ -7,9 +34,80 @@ const React = require('react');
...
+      this.setState({
+        campaigns: campaigns,
+        current_campaign_index: 0
+      });
+    });
+  }
+
+  componentWillUnmount() {
+    if (this.serverRequest) {
+      this.serverRequest.abort();;
```
Comment (weerd) [inline lib/themes/dosomething/paraneue_dosomething/js/components/FeatureCard.js]: you _really_ want this statement to end eh? double semi-colons!! `;;`

File: lib/themes/dosomething/paraneue_dosomething/js/components/FeatureCard.js [hunk 5]
```diff
@@ -7,9 +34,80 @@ const React = require('react');
...
+    }
+  }
+
+  signup() {
+    let campaign = this.state.campaigns[this.state.current_campaign_index];
```
Comment (weerd) [inline lib/themes/dosomething/paraneue_dosomething/js/components/FeatureCard.js]: can probs change this to `const` since you're not reassigning within the block scope.

File: lib/themes/dosomething/paraneue_dosomething/js/components/FeatureCard.js [hunk 6]
```diff
@@ -7,9 +34,80 @@ const React = require('react');
...
+    }
+
+    this.setState({
+      current_campaign_index: newIndex,
+      signup_animation: ''
+    });
   }
 
   render() {
+    let campaign = this.state.campaigns[this.state.current_campaign_index];
```
Comment (weerd) [inline lib/themes/dosomething/paraneue_dosomething/js/components/FeatureCard.js]: `const`?

File: lib/themes/dosomething/paraneue_dosomething/js/onboarding.js [hunk 7]
```diff
@@ -20,8 +18,12 @@ $(document).ready(function() {
   const disabledByLanguage = ['mx', 'br'].indexOf(campaign.language.language_code) != -1;
   let localStorageKey = null;
 
-  if (campaign && user && !disabledByLanguage && !Modal.isOpen()) {
-    localStorageKey = `onboarding:campaign=${campaign.id}-user=${user.info.id}-version=${ONBOARDING_VERSION}`;
+  if (Modal.isOpen()) {
```
Comment (weerd) [inline lib/themes/dosomething/paraneue_dosomething/js/onboarding.js:21]: So if there's an open modal we don't show Onboarding?

File: lib/themes/dosomething/paraneue_dosomething/scss/content/_onboarding.scss [hunk 8]
```diff
@@ -2,10 +2,10 @@
   background-color: $purple;
   color: $white;
   position: relative;
+  max-width: 1440px;
```
Comment (weerd) [inline lib/themes/dosomething/paraneue_dosomething/scss/content/_onboarding.scss]: good call on this, but can you revise it to do it this way instead:

``` css
.onboarding {
  > .wrapper {
    background-color: $purple;
    color: $white;
    max-width: 1440px;
    margin: 0 auto;
    padding: $base-spacing 0 $section-spacing;
    position: relative;
  }
...
}
```

remove rules from the parent `.onboarding` and put them in the `.wrapper` instead and in alphabetical order 😉

File: lib/themes/dosomething/paraneue_dosomething/js/onboarding.js (hunk 7 above)
Comment (itsjoekent) [inline lib/themes/dosomething/paraneue_dosomething/js/onboarding.js:21]: yus

File: lib/themes/dosomething/paraneue_dosomething/scss/content/_onboarding.scss (hunk 8 above)
Comment (itsjoekent) [inline lib/themes/dosomething/paraneue_dosomething/scss/content/_onboarding.scss]: okie dokes

**Comment by itsjoekent** [comment]
tldr: USE MORE CONST

**Comment by weerd** [comment]
lol. "use more `const`" just when the variable defined isn't going to change :)

**Comment by itsjoekent** [comment]
@weerd 💥 i think i got em all

**Comment by weerd** [comment]
👍

**Comment by itsjoekent** [comment]
🚂 🚋 🚋
